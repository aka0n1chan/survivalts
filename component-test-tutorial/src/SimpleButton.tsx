import { useState } from 'react';

//引数なしでJSX.Elementを返す関数という型定義を与えている
export const SimpleButton: () => JSX.Element = () => {
  const [state, setState] = useState(false);
  const handleClick = () => {
    setState((prevState) => !prevState);
  };
  return <button onClick={handleClick}>{state ? 'ON' : 'OFF'}</button>;
};
