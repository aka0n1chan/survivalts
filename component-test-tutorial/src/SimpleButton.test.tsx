import { render, screen } from '@testing-library/react';
import { SimpleButton } from './SimpleButton';
import userEvent from '@testing-library/user-event';

test('ボタンをクリックするとON/OFFの表示が切り替わる', async () => {
  const user = userEvent.setup();
  //ボタンの描画
  render(<SimpleButton />);
  //ボタンのDOMを取得
  const simpleButton = screen.getByRole('button');
  //アサーション
  expect(simpleButton).toHaveTextContent('OFF');
  //クリックイベントを発生させる
  await user.click(simpleButton);
  //アサーション
  expect(simpleButton).toHaveTextContent('ON');
});
