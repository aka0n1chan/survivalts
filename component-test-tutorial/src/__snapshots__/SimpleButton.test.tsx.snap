// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`描画されてすぐはOFFと表示 1`] = `
<div>
  <button>
    OFF
  </button>
</div>
`;
